presto-main/src/main/java/com/facebook/presto/operator/scalar/TDigestFunctions.java

```diff
@@ -86,4 +104,40 @@ public static Slice scaleTDigestDouble(@SqlType("tdigest(double)") Slice input,
         digest.scale(scale);
         return digest.serialize();
     }
+
+    @ScalarFunction(value = "destructure_tdigest", visibility = EXPERIMENTAL)
+    @Description("Return the raw TDigest, including arrays of centroid means and weights, as well as min, max, sum, count, and compression factor.")
+    @SqlType("row(centroid_means array(double), centroid_weights array(integer), compression double, min double, max double, sum double, count integer)")
+    public static Block destructureTDigest(@SqlType("tdigest(double)") Slice input)
+    {
+        TDigest tDigest = createTDigest(input);
+
+        PageBuilder pageBuilder = new PageBuilder(ImmutableList.of(TDIGEST_CENTROIDS_ROW_TYPE));
```

Comment:
You don't really need to create a `PageBuilder` here. Please directly create a `RowBlockBuilder`. You can use `RowType.createBlockBuilder()`

Reply:
Thanks for the suggestion. Addressed in db91cce971fce65dd097de1a089ec92f49f9be8a